### PR TITLE
Cleanup test infra after e2e tests

### PR DESCRIPTION
**Pull request by dangula**
fix for 629 and 640
We are calling clean up before starting a new test suite run so any residue of test infra from older runs are removed.
Also cleaning up test images and adding file system test.

### CODE REVIEW

File: e2e/tests/smoke/file_storage_smoke_test.go [hunk 1]
```diff
@@ -42,12 +43,6 @@ func (suite *FileSystemTestSuite) SetupTest() {
 
 	require.NotEmpty(suite.T(), suite.rookTag, "RookTag parameter is required")
 
-}
-
-func (suite *FileSystemTestSuite) TestFileStorage_SmokeTest() {
-	suite.T().Skip("Skipping test : existing issue https://github.com/rook/rook/issues/612")
```
Comment (travisn) [inline e2e/tests/smoke/file_storage_smoke_test.go]: let's scope this PR to only the changes needed to get the tests working again. we can save enabling the file tests for another PR

File: e2e/README.md [hunk 2]
```diff
@@ -1,30 +1,32 @@
 # Rook Test Framework
  
 The Rook Test Framework is a Go project that that uses a Docker container with Kubernetes
-tools and Docker pre-installed to quickly create a 3 node Kubernetes cluster using Docker in 
+tools and Docker pre-installed to quickly create a three node Kubernetes cluster using Docker in 
 Docker (DinD) techniques, installs Rook and then runs automation tests to verify its functionality.
  
 ##Requirements
```
Comment (travisn) [inline e2e/README.md]: a space is needed after the double hash in order to show as a title

File: e2e/tests/smoke/file_storage_smoke_test.go (hunk 1 above)
Comment (dangula) [inline e2e/tests/smoke/file_storage_smoke_test.go]: ignoring this test again, I will enable it as part of #633

File: e2e/framework/manager/rook_test_infra.go [hunk 3]
```diff
@@ -196,8 +198,24 @@ func (r *rookTestInfraManager) copyImageToNode(containerId string, imageName str
 	return nil
 }
 
+func (r *rookTestInfraManager) configureSetupStatus(status bool) {
```
Comment (travisn) [inline e2e/framework/manager/rook_test_infra.go]: if you're just going to set a flag, i think the calling code would be more readable if you didn't have these one-line helper methods `configureSetupStatus` and `confugreRookInstalledStatus`. Just assign `r.isSetup = true` directly.

File: e2e/scripts/smoke_test.sh [hunk 4]
```diff
@@ -71,14 +72,22 @@ rook_infra::gather_results() {
...
-{ #try
+    #Run clean up tests that runs down on dind script
+    docker exec -t ${id} /bin/bash -c \
+        "go test -timeout 1200s -run TestRookInfraCleanUp  ${git_test_directory} --rook_platform=${rook_platform} --k8s_version=${k8s_version} --rook_version=${tag_name} -v 2>&1"
 
-    if [ -z "$1" ]; then
+    echo Removing rook-test-framework container and images...
+    docker kill ${id} || true
+    docker rm ${id} || true
+    docker rmi $(docker images|grep 'rook-test\|kubeadm-dind-cluster\|ceph/base') > /dev/null 2>&1 || true
```
Comment (travisn) [inline e2e/scripts/smoke_test.sh]: does this grep only output a single image? if multiple images it's likely going to fail. You might need to pipe the results of the grep to `xargs docker rmi`

File: e2e/tests/smoke/file_storage_smoke_test.go [hunk 5]
```diff
@@ -58,50 +53,58 @@ func (suite *FileSystemTestSuite) TestFileStorage_SmokeTest() {
...
 	require.Nil(suite.T(), fsd_err)
 	//Delete is not actually deleting filesystem
 	suite.T().Log("File system deleted")
 }
 
-func fileSmokecleanUp() {
-	sc, _ := CreateSmokeTestClient(enums.Kubernetes)
-	sc.UnmountFileStorage()
-	sc.DeleteFileStorage()
+func fileSmokecleanUp(h *SmokeTestHelper) {
```
Comment (travisn) [inline e2e/tests/smoke/file_storage_smoke_test.go]: nit: capitalize `clean`

**Pull request merged**